pkg/specgen/generate/container.go

```diff
@@ -61,6 +63,62 @@ func getImageFromSpec(ctx context.Context, r *libpod.Runtime, s *specgen.SpecGen
 	return image, resolvedName, inspectData, err
 }
 
+func applyHealthCheckOverrides(s *specgen.SpecGenerator, healthCheckFromImage *manifest.Schema2HealthConfig) error {
+	overrideHealthCheckConfig := s.HealthConfig
+	s.HealthConfig = healthCheckFromImage
+
+	if s.HealthConfig == nil {
+		return nil
+	}
+
+	disableInterval := false
+	if overrideHealthCheckConfig != nil {
+		disableInterval = slices.Contains(overrideHealthCheckConfig.Test, define.HealthConfigTestOverrideDisableInterval)
```

Comment:
These special values still feel totally wrong and I don't get why they ar eneeded at all, this should just work without it.

Reply:
Because if you set `--health-interval disable`, it will be converted to `0` in the hc configuration. In the hc config `0` means the default value, so in the next steps this value will be overwritten by the default value. I want to avoid this override.

Reply:
Odd, why is "disable" even a thing for interval?
`--health-interval disable` disabled the healtcheck? We also have --no-healthcheck for this.

A duration can be negative would it be possible for us to pass `-1` instead to mean disable? I guess the issue here is that this would break the client <-> server interface, e.g. if you use a new client against and old server this will likely result in something unexpected. However the same concern is for your special value in the cmd as the server would also not understand it.

I am not sure what is best here.

Reply:
Zero was always represented as a forbidden interval. So I'll leave it at that, no one is complaining that zero for an interval doesn't mean it's the default value. But for command line processing, use `-1` and then set it to `0`. Maybe `//TODO: v6.0`?

The `--health-interval disable` will disable the timer, but you can still run it with `podman healthcheck run`. The `--no-healthcheck` disables the entire healthcheck configuration, so you cannot run it with `podman healthcheck run`.